### PR TITLE
Slackbot whitelisting improvements

### PR DESCRIPTION
**Pull request by ajvb**
This PR includes two whitelisting improvements.

The first is adding support for a `never` expiration value. This will
set the TTL to ten years from now.

The second is that when an IP is whitelisted, the slackbot will iterate
through all provided iprepd urls and send DELETE requests.

As well, this PR switches slackbot-background to using sops and includes
a new `Configuration` struct that abstracts this (so that all of the
other cloudfunctions can start using it).

This was done in this PR as trying to shove multiple sets of iprepd
urls/api key combinations into a single environment variable was getting
finicky.

Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1557140

### CODE REVIEW

File: slackbot-background/background.go [hunk 1]
```diff
@@ -23,104 +22,68 @@ const (
 	DURATION_DOC                       = "FoxsecBot uses Go's time.ParseDuration internally " +
 		"with some custom checks. Examples: '72h' or '2h45m'. " +
 		"Valid time units are 'm' and 'h'. If you omit a duration, " +
-		"the default (24 hours) is used. If your duration is under 5 minutes, it is increased to 5 minutes."
+		"the default (24 hours) is used. If your duration is under " +
+		"5 minutes, it is increased to 5 minutes. If do not want the " +
```
Comment (ameihm0912) [inline slackbot-background/background.go]: typo here

File: slackbot-background/utils.go [hunk 2]
```diff
@@ -86,3 +92,24 @@ func sendSlackCallback(msg *slack.Msg, responseUrl string) error {
 func isAlertConfirm(callbackId string) bool {
 	return strings.HasPrefix(callbackId, "alert_confirmation")
 }
+
+func deleteIpFromIprepd(ip string) error {
+	client := http.Client{Timeout: time.Second * 10}
+	for _, iprepdInstance := range config.IprepdInstances {
+		log.Infof("Sending DELETE request to %s for %s", iprepdInstance.URL, ip)
+
+		req, err := http.NewRequest("DELETE", iprepdInstance.URL+"/"+ip, nil)
```
Comment (ameihm0912) [inline slackbot-background/utils.go:101]: This is making use of the legacy API endpoint; might be good to update this to use the new one,

```go
req, err := http.NewRequest("DELETE", iprepdInstance.URL + "/type/ip/" + ip, nil)
```

File: slackbot-background/handlers.go [hunk 3]
```diff
@@ -50,8 +50,13 @@ func handleWhitelistCmd(ctx context.Context, cmd common.SlashCommandData, db *co
 		return msg, err
 	}
 
+	err = deleteIpFromIprepd(ip.String())
```
Comment (ameihm0912) [inline slackbot-background/handlers.go:53]: Do we verify the format of this string to ensure it is actually an address? I didn't see that anywhere but might have missed it.

**Review by ameihm0912 — APPROVED** [review]
Looks good, few notes for consideration

File: slackbot-background/utils.go (hunk 2 above)
Comment (ajvb) [reply slackbot-background/utils.go:101]: @ameihm0912 is iprepd 0.8 getting deployed within GCP here shortly? If not, happy to come back and update this once that is done.

Comment (ameihm0912) [reply slackbot-background/utils.go:101]: ahhh right, good call -- yeah let's leave as is then for now until we have coverage there

File: slackbot-background/handlers.go (hunk 3 above)
Comment (ajvb) [reply slackbot-background/handlers.go:53]: Yeah, this is done within the parse function:

https://github.com/mozilla-services/foxsec-pipeline-contrib/blob/a99ae9291b222c89fc1175b573d1be40fc1560b0/slackbot-background/utils.go#L40-L48

Comment (ameihm0912) [reply slackbot-background/handlers.go:53]: 👍

**Pull request merged**